File: Akshit_22b2531/week1/coin_piles.py
t = int(input())
for _ in range(t):
    a, b = map(int, input().split())
    total = a + b
    # Both must be divisible by 3 and neither pile can have more than double the other
    if total % 3 == 0 and min(a, b) * 2 >= max(a, b):
        print("YES")
    else:
        print("NO")

File: Parakh_24b0729/week-1/coinpiles.cpp
#include <iostream>
using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        long long a, b;
        cin >> a >> b;
        // Both a and b must be divisible by 3 together, and neither can be more than double the other
        if ((a + b) % 3 == 0 && 2 * a >= b && 2 * b >= a) {
            cout << "YES\n";
        } else {
            cout << "NO\n";
        }
    }
    return 0;
}